src/components/ModalPayment/SquareModal.tsx

```diff
@@ -61,117 +56,113 @@ const ModalPayment = ({
     return makeSquarePayment( nonce, payment, buyer )
       .then((res) => {
         if (res.status === 200) {
-          setIsShown(true)
-        } else {
-          setIsShown(false)
+          dispatch({type: SET_MODAL_VIEW, payload: 2})
+        } 
+      })
+      .catch((err) => {
+        if( err.response
+          && err.response.data
+          && err.response.data.errors
+          && err.response.data.errors.length > 0){          
+          const responseErrors = err.response.data.errors
+          const newErrors = errorMessages.length > 0 ? [...errorMessages,responseErrors.map((error: { detail: string }) => error.detail) ] : responseErrors.map((error: { detail: string }) => error.detail)
+          setErrorsMessages(newErrors)
         }
       })
   }
 
   const applicationId = process.env.REACT_APP_SQUARE_APPLICATION_ID ? process.env.REACT_APP_SQUARE_APPLICATION_ID : ""
   const locationId = process.env.REACT_APP_SQUARE_SANDBOX_LOCATION_ID ? process.env.REACT_APP_SQUARE_SANDBOX_LOCATION_ID : ""
 
+  const canSubmit = isChecked && name.length > 0 && email.length > 0;
```

Comment:
also if there are no Square errors

Reply:
yeah we'll likely want to expand this to test email validity, there are some regex ways to do this or we can look at other options